generate-total-list.js

```diff
@@ -0,0 +1,39 @@
+const addProductButton = document.querySelector(".add-product-form__button");
+const chosenProductAmountValue = document.querySelector(".add-product-form__input").value;
+const chosenProductListWrapper = document.querySelector(".chosen-product-list__wrapper");
+
+const addProduct = event => {
+    let currentProductName = document.querySelector(".add-product-form__dropdown").value;
+    
+    products.forEach(product =>{
```

Comment:
Spoko, że to wykombinowałeś, można to jeszcze uprościć i zrobić bardziej semantycznie.
```
const currentProductNameValue = document.querySelector(
  '.add-product-form__dropdown',
).value;

const currentProduct = products.find(
  product => product.value === currentProductNameValue,
);
```
`products.find(...)` zwróci po prostu obiekt, dla którego funkcja podana do `find` zwróci `true`.

Potem możesz zamiast `currentProductName` dać `currentProduct.name`.